iradix_test.go

```diff
@@ -212,6 +211,145 @@ func TestDelete(t *testing.T) {
 	}
 }
 
+func TestDeletePrefix(t *testing.T) {
+
+	type exp struct {
+		treeNodes   []string
+		prefix      string
+		expectedOut []string
+	}
+
+	//various test cases where DeletePrefix should succeed
+	cases := []exp{
+		{[]string{"", "test/test1", "test/test2", "test/test3", "R", "RA"}, "test", []string{"", "R", "RA"}},
+		{[]string{"", "test", "test/test1", "test/test2", "test/test3", "test/testAAA", "R", "RA"}, "test", []string{"", "R", "RA"}},
+		{[]string{"", "test/test1", "test/test2", "test/test3", "test/testAAA", "R", "RA"}, "test/test", []string{"", "R", "RA"}},
+		{[]string{"", "AB", "ABC", "AR", "R", "RA"}, "AR", []string{"", "AB", "ABC", "R", "RA"}},
+	}
+
+	for _, testCase := range cases {
+		verifyPrefixDelete(t, testCase.treeNodes, testCase.prefix, testCase.expectedOut)
+	}
+
+	//verify deleting a prefix that doesn't exist
+	verifyInvalidPrefix(t, []string{"", "test/test1", "test/test2"}, "XXXX")
```

Comment:
It's a little weird this is on the end here. You could verify deleting a "nope" prefix for each of the test cases and re-verify it matches the expected, so this becomes part of each test case.